### PR TITLE
[v2.0.3] Swagger Configuration for Auth Server API documentation

### PR DESCRIPTION
**Pull request by dhanyak-btc**
1. Added swagger configuration to generate API documentation
2. Added test to create `openapi.json `under `/fda-mystudies/documentation/API/auth-server `

API Documentation URL: ` http://<HOST>:<PORT>/auth-server/swagger-ui.html#` or copy the content of this [file ](https://github.com/GoogleCloudPlatform/fda-mystudies/blob/1a9de667260a897c277725e26be6d8b734758a2d/documentation/API/auth-server/openapi.json) into [Swagger Editor](https://editor.swagger.io/)

### CODE REVIEW

File: auth-server/oauth-scim-service/src/main/java/com/google/cloud/healthcare/fdamystudies/oauthscim/controller/OAuthController.java [hunk 1]
```diff
@@ -60,6 +66,11 @@
 
   @Autowired private AuthScimAuditHelper auditHelper;
 
+  @ApiOperation(
+      notes =
+          "Refer [The OAuth 2.0 Token Endpoint](https://www.ory.sh/hydra/docs/reference/api/#the-oauth-20-token-endpoint) for request and response details ",
+      value =
+          "Get access token and refresh token based on grant type. Refer ORY Hydra REST API documentation for request and response details")
```
[the remark this answers is not in the document]
Comment (dhanyak-btc) [reply auth-server/oauth-scim-service/src/main/java/com/google/cloud/healthcare/fdamystudies/oauthscim/controller/OAuthController.java]: Fixed this issue

File: auth-server/oauth-scim-service/src/main/java/com/google/cloud/healthcare/fdamystudies/oauthscim/controller/OAuthController.java [hunk 2]
```diff
@@ -26,6 +26,8 @@
 import com.google.cloud.healthcare.fdamystudies.mapper.AuditEventMapper;
 import com.google.cloud.healthcare.fdamystudies.oauthscim.common.AuthScimAuditHelper;
 import com.google.cloud.healthcare.fdamystudies.oauthscim.service.OAuthService;
+import io.swagger.annotations.Api;
```
Comment (moschetti) [inline auth-server/oauth-scim-service/src/main/java/com/google/cloud/healthcare/fdamystudies/oauthscim/controller/OAuthController.java:29]: update copyright header to 2021

File: auth-server/oauth-scim-service/src/main/java/com/google/cloud/healthcare/fdamystudies/oauthscim/controller/OAuthController.java [hunk 3]
```diff
@@ -60,6 +66,13 @@
 
   @Autowired private AuthScimAuditHelper auditHelper;
 
+  @ApiOperation(
+      notes =
+          "Refer [The OAuth 2.0 Token Endpoint](https://www.ory.sh/hydra/docs/reference/api/#the-oauth-20-token-endpoint)"
```
Comment (moschetti) [inline auth-server/oauth-scim-service/src/main/java/com/google/cloud/healthcare/fdamystudies/oauthscim/controller/OAuthController.java:71]: `Refer to the ...`

File: auth-server/oauth-scim-service/src/main/java/com/google/cloud/healthcare/fdamystudies/oauthscim/controller/OAuthController.java [hunk 4]
```diff
@@ -60,6 +66,13 @@
...
+              + " for request and response details ",
+      value =
+          "Get access token and refresh token based on grant type. "
+              + "Refer ORY Hydra REST API documentation for request and response details")
```
Comment (moschetti) [inline auth-server/oauth-scim-service/src/main/java/com/google/cloud/healthcare/fdamystudies/oauthscim/controller/OAuthController.java:75]: same as above

File: auth-server/oauth-scim-service/src/test/java/com/google/cloud/healthcare/fdamystudies/SwaggerGeneratorTest.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2020 Google LLC
```
Comment (moschetti) [inline auth-server/oauth-scim-service/src/test/java/com/google/cloud/healthcare/fdamystudies/SwaggerGeneratorTest.java:2]: update to 2021

File: common-modules/common-service/src/main/java/com/google/cloud/healthcare/fdamystudies/config/SwaggerConfig.java [hunk 6]
```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2020 Google LLC
```
Comment (moschetti) [inline common-modules/common-service/src/main/java/com/google/cloud/healthcare/fdamystudies/config/SwaggerConfig.java:2]: 2021

File: common-modules/common-service/src/main/java/com/google/cloud/healthcare/fdamystudies/controller/HealthController.java [hunk 7]
```diff
@@ -8,17 +8,50 @@
 
 package com.google.cloud.healthcare.fdamystudies.controller;
 
+import com.google.cloud.healthcare.fdamystudies.beans.BaseResponse;
```
Comment (moschetti) [inline common-modules/common-service/src/main/java/com/google/cloud/healthcare/fdamystudies/controller/HealthController.java:11]: update copyright to 2021

File: common-modules/common-tests/src/main/java/com/google/cloud/healthcare/fdamystudies/common/BaseMockIT.java [hunk 8]
```diff
@@ -25,14 +25,20 @@
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
```
Comment (moschetti) [inline common-modules/common-tests/src/main/java/com/google/cloud/healthcare/fdamystudies/common/BaseMockIT.java:28]: update copyright to 2021

File: auth-server/oauth-scim-service/src/main/java/com/google/cloud/healthcare/fdamystudies/oauthscim/controller/OAuthController.java (hunk 2 above)
Comment (dhanyak-btc) [reply auth-server/oauth-scim-service/src/main/java/com/google/cloud/healthcare/fdamystudies/oauthscim/controller/OAuthController.java:29]: Fixed

File: auth-server/oauth-scim-service/src/main/java/com/google/cloud/healthcare/fdamystudies/oauthscim/controller/OAuthController.java (hunk 3 above)
Comment (dhanyak-btc) [reply auth-server/oauth-scim-service/src/main/java/com/google/cloud/healthcare/fdamystudies/oauthscim/controller/OAuthController.java:71]: Fixed

File: auth-server/oauth-scim-service/src/main/java/com/google/cloud/healthcare/fdamystudies/oauthscim/controller/OAuthController.java (hunk 4 above)
Comment (dhanyak-btc) [reply auth-server/oauth-scim-service/src/main/java/com/google/cloud/healthcare/fdamystudies/oauthscim/controller/OAuthController.java:75]: Fixed

File: auth-server/oauth-scim-service/src/test/java/com/google/cloud/healthcare/fdamystudies/SwaggerGeneratorTest.java (hunk 5 above)
Comment (dhanyak-btc) [reply auth-server/oauth-scim-service/src/test/java/com/google/cloud/healthcare/fdamystudies/SwaggerGeneratorTest.java:2]: Fixed

File: common-modules/common-service/src/main/java/com/google/cloud/healthcare/fdamystudies/config/SwaggerConfig.java (hunk 6 above)
Comment (dhanyak-btc) [reply common-modules/common-service/src/main/java/com/google/cloud/healthcare/fdamystudies/config/SwaggerConfig.java:2]: Fixed

File: common-modules/common-service/src/main/java/com/google/cloud/healthcare/fdamystudies/controller/HealthController.java (hunk 7 above)
Comment (dhanyak-btc) [reply common-modules/common-service/src/main/java/com/google/cloud/healthcare/fdamystudies/controller/HealthController.java:11]: Fixed

**Comment by dhanyak-btc** [comment]
@moschetti Fixed review comments, please re-review and approve the PR.

**Review by moschetti — APPROVED** [review]

**Pull request merged**